com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -127,13 +176,43 @@ internal struct AgentParameters
         /// <summary>
         /// The maximum number of steps the agent takes before being done.
         /// </summary>
+        /// <value>The maximum steps for an agent to take before it resets; or 0 for
+        /// unlimited steps.</value>
         /// <remarks>
-        /// If set to 0, the agent can only be set to done programmatically (or
-        /// when the Academy is done).
-        /// If set to any positive integer, the agent will be set to done after
-        /// that many steps. Note that setting the max step to a value greater
-        /// than the academy max step value renders it useless.
+        /// The max step value determines the maximum length of an agent's episodes.
+        /// Set to a positive integer to limit the episode length to that many steps.
+        /// Set to 0 for unlimited episode length. 
+        ///
+        /// When an episode ends and a new one begins, the Agent object's
+        /// <seealso cref="OnEpisodeBegin"/> function is called. You can implement
+        /// <see cref="OnEpisodeBegin"/> to reset the agent or remove it from the
```

Comment:
Same comment as above. It seems this information here is redundant. Maybe we should make sure we say everything only once. This will make it easier to change the documentation in the future.